### PR TITLE
(Test) O3-3754: Add unit tests for the Items table

### PR DESCRIPTION
**Pull request by TrevorAntony**
## Summary
Consists of tests that ensure that the component renders correctly, 
Verifies that the component renders with the initial state and UI elements are correctly displayed.
Tests that the skeleton loader is displayed when isLoading is true.
Tests that the table rows are rendered correctly based on the items prop.
Tests the search input to ensure it updates the state and triggers the search function correctly. 
Tests the pagination controls to ensure they update the current page and page size correctly.
Tests that clicking the refresh button triggers the handleRefresh function.
Tests that clicking the edit button on a row triggers the launchAddOrEditDialog function with the correct arguments.

## Related Issue
https://openmrs.atlassian.net/browse/O3-3754

### CODE REVIEW

**Comment by TrevorAntony** [comment]
@CynthiaKamau @ODORA0 @hadijahkyampeire

**Review by jabahum — COMMENTED** [review]
@denniskigen did some clean ups in this PR https://github.com/openmrs/openmrs-esm-stock-management/pull/220/files can try i look through his changes to avoid this

**Comment by ODORA0** [comment]
@TrevorAntony What we can do here is change occurrences of the fireEvent and adopt userEvent to simulate user interaction. Declare in each test suite the userEvent, see [here](https://github.com/openmrs/openmrs-esm-stock-management/pull/220/files#diff-40ce954cef88ff808f7ebae40b1b49bb4f2e2d4e656bb1b1282b23f055681a7eR111)  and replace anything like [this](https://github.com/openmrs/openmrs-esm-stock-management/pull/220/files#diff-40ce954cef88ff808f7ebae40b1b49bb4f2e2d4e656bb1b1282b23f055681a7eL116) to [this](https://github.com/openmrs/openmrs-esm-stock-management/pull/220/files#diff-40ce954cef88ff808f7ebae40b1b49bb4f2e2d4e656bb1b1282b23f055681a7eR116). Just to keep consistency with the convention

File: src/stock-items/stock-items-table.test.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    (useStockItemsPages as jest.Mock).mockReturnValue(mockUseStockItemsPages);
+  });
+
+  it('renders initial state and UI elements correctly', async () => {
+    render(<StockItemsTableComponent />);
+    expect(screen.getByText('panelDescription')).toBeInTheDocument();
+    expect(screen.getByRole('searchbox')).toBeInTheDocument();
+    
+    const menuButton = screen.getByTestId('stock-items-menu');
+    fireEvent.click(menuButton);
```
Comment (jabahum) [inline src/stock-items/stock-items-table.test.tsx:61]: missed this

File: src/stock-items/stock-items-table.test.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    render(<StockItemsTableComponent />);
+    expect(screen.getByText('Test Item 0')).toBeInTheDocument();
+    expect(screen.getAllByText('drug').length).toBeGreaterThan(0);
+    expect(screen.getAllByText('other').length).toBeGreaterThan(0);
+  });
+
+  it('updates search input and triggers search function', async () => {
+    render(<StockItemsTableComponent />);
+    const searchInput = screen.getByRole('searchbox');
+    fireEvent.change(searchInput, { target: { value: 'test search' } });
```
Comment (jabahum) [inline src/stock-items/stock-items-table.test.tsx:86]: this too

File: src/stock-items/stock-items-table.test.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    await waitFor(() => {
+      expect(mockUseStockItemsPages.setSearchString).toHaveBeenCalledWith('test search');
+    }, { timeout: 2000 });
+  });
+
+  it('updates pagination when page or page size changes', () => {
+    render(<StockItemsTableComponent />);
+    const nextPageButton = screen.getByLabelText('Next page');
+    fireEvent.click(nextPageButton);
```
Comment (jabahum) [inline src/stock-items/stock-items-table.test.tsx:95]: this too

File: src/stock-items/stock-items-table.test.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    expect(mockUseStockItemsPages.setCurrentPage).toHaveBeenCalled();
+
+    const pageSizeSelect = screen.getByLabelText('Items per page:');
+    fireEvent.change(pageSizeSelect, { target: { value: '20' } });
```
Comment (jabahum) [inline src/stock-items/stock-items-table.test.tsx:99]: this too

File: src/stock-items/stock-items-table.test.tsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    expect(mockUseStockItemsPages.setPageSize).toHaveBeenCalledWith(20);
+  });
+
+  it('triggers handleRefresh when refresh button is clicked', async () => {
+    render(<StockItemsTableComponent />);
+    
+    const menuButton = screen.getByTestId('stock-items-menu');
+    expect(menuButton).toBeInTheDocument();
+    fireEvent.click(menuButton);
```
Comment (jabahum) [inline src/stock-items/stock-items-table.test.tsx:108]: this too

File: src/stock-items/stock-items-table.test.tsx [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+  
+    const refreshButton = await screen.findByText('Refresh');
+    expect(refreshButton).toBeInTheDocument();
+    fireEvent.click(refreshButton);
```
Comment (jabahum) [inline src/stock-items/stock-items-table.test.tsx:112]: this too

File: src/stock-items/stock-items-table.test.tsx [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    await waitFor(() => {
+      expect(handleMutate).toHaveBeenCalledWith(expect.stringContaining('/stockmanagement/stockitem'));
+    });
+  });
+
+  it('triggers launchAddOrEditDialog when edit button is clicked', () => {
+    render(<StockItemsTableComponent />);
+    const editButtons = screen.getAllByLabelText('Edit Stock Item');
+    fireEvent.click(editButtons[0]);
```
Comment (jabahum) [inline src/stock-items/stock-items-table.test.tsx:121]: finally this

File: src/stock-items/stock-items-table.test.tsx (hunk 1 above)
Comment (TrevorAntony) [reply src/stock-items/stock-items-table.test.tsx]: pushing these as well

File: src/stock-items/stock-items-table.test.tsx (hunk 7 above)
Comment (jabahum) [reply src/stock-items/stock-items-table.test.tsx]: this too

**Comment by TrevorAntony** [comment]
@jabahum kindly review

**Review by jabahum — APPROVED** [review]

**Pull request merged**